### PR TITLE
Mecanum drive

### PR DESCRIPTION
**Pull request by Rkortlandt**
Combine the MecanumDrive with Dev

### CODE REVIEW

**Comment by Rkortlandt** [comment]
The Code Factor is just mad at some comments nothing huge they can be removed after merge

File: src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  private final MecanumDrive m_MecanumDrive = new MecanumDrive(m_frontLeftMotor, m_rearLeftMotor, m_frontRightMotor, m_rearRightMotor);
+  
+  public void setDriveMotorControllersVolts(MecanumDriveMotorVoltages volts) {
+    m_frontLeftMotor.setVoltage(volts.frontLeftVoltage);
+    m_rearLeftMotor.setVoltage(volts.rearLeftVoltage);
+    m_frontRightMotor.setVoltage(volts.frontRightVoltage);
+    m_rearRightMotor.setVoltage(volts.rearRightVoltage);
+  }
+
+  // private final Encoder m_frontLeftEncoder =
```
Comment (SeanErn) [inline src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java:77]: Remove these unused comments please

File: src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  }
+
+  // public void resetEncoders() {
+  //   m_frontLeftEncoder.reset();
+  //   m_rearLeftEncoder.reset();
+  //   m_frontRightEncoder.reset();
+  //   m_rearRightEncoder.reset();
+  // }
+
+  public void zeroHeading() {
```
Comment (SeanErn) [inline src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java:166]: should be inside the gyro subsystem

File: src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+Translation2d m_frontRightLocation = new Translation2d(0.381, -0.381);
+Translation2d m_backLeftLocation = new Translation2d(-0.381, 0.381);
+Translation2d m_backRightLocation = new Translation2d(-0.381, -0.381);
+
+// Creating my kinematics object using the wheel locations.
+MecanumDriveKinematics m_kinematics = new MecanumDriveKinematics(
+  m_frontLeftLocation, m_frontRightLocation, m_backLeftLocation, m_backRightLocation
+);
+
+// Creating my odometry object from the kinematics object and the initial wheel positions.
```
Comment (SeanErn) [inline src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java:118]: move all refrences to the gyro to the gyro subsys

File: src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java (hunk 1 above)
Comment (SeanErn) [inline src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java:77]: remove any unused comments

File: src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+   * Returns the turn rate of the robot.
+   *
+   * @return The turn rate of the robot, in degrees per second
+   */
+  public double getTurnRate() {
+    return -m_gyro.getRate();
+  }
+
+  /** Initializes Shuffleboard motor objects */
+  public void initShuffleboard(){
```
Comment (SeanErn) [inline src/main/java/frc/robot/ChargedUp/MecanumDrive/Subsys_MecanumDrive.java:190]: broken, sometimes causes robot crash if used. Remove or fix

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -45,12 +49,12 @@ public class RobotContainer {
   private final SubSys_LimeLight limeLightSubSys = new SubSys_LimeLight();
 
   // ---- Drive Subsystem (Swerve)
-  public final SubSys_DriveTrain driveSubSys = new SubSys_DriveTrain(gyroSubSys);
```
Comment (SeanErn) [inline src/main/java/frc/robot/RobotContainer.java:48]: Don't remove the SubSys_DriveTrain, use a sendable chooser to select what drive you want with smart dashboard

File: src/main/java/frc/robot/RobotContainer.java [hunk 6]
```diff
@@ -110,32 +129,32 @@ public RobotContainer() {
...
    * passing it to a {@link edu.wpi.first.wpilibj2.command.button.JoystickButton}.
    */
-  private void configureButtonBindings() {
+   private void configureButtonBindings() {
  
     // Gyro Reset Command Button
-    driverStation.GyroResetButton.onTrue(
-        new InstantCommand(driveSubSys::zeroGyro, driveSubSys));
-  }
+    // driverStation.GyroResetButton.onTrue(
```
Comment (SeanErn) [inline src/main/java/frc/robot/RobotContainer.java:144]: Use long comments please

File: src/main/java/frc/robot/ChargedUp/MecanumDrive/Const_MecanumDrive.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    //TODO: TUNE PID
+    /** Drive PID  LINEAR */
+    public static final double k_driveLinKp = 0.1;
+    public static final double k_driveLinKi = 0;
+    public static final double k_driveLinKd = 0;
+    public static final double k_driveAngKp = 0.1;
+    public static final double k_driveAngKi = 0;
+    public static final double k_driveAngKd = 0;
+
+    /** Encoders */
```
Comment (SeanErn) [inline src/main/java/frc/robot/ChargedUp/MecanumDrive/Const_MecanumDrive.java:20]: Ensure these are actually the correct encoder ports

File: src/main/java/frc/robot/ChargedUp/MecanumDrive/Const_MecanumDrive.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    public static final int[] k_RearLeftEncoderPorts = new int[] {2, 3};
+    public static final int[] k_FrontRightEncoderPorts = new int[] {4, 5};
+    public static final int[] k_RearRightEncoderPorts = new int[] {6, 7};
+
+    public static final boolean k_FrontLeftEncoderReversed = false;
+    public static final boolean k_RearLeftEncoderReversed = true;
+    public static final boolean k_FrontRightEncoderReversed = false;
+    public static final boolean k_RearRightEncoderReversed = true;
+
+    public static final double k_TrackWidth = 0.5;
```
Comment (SeanErn) [inline src/main/java/frc/robot/ChargedUp/MecanumDrive/Const_MecanumDrive.java:31]: Make track width/wheelbase correct and have one for dev bot and prod bot

**Review by SeanErn — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java (hunk 6 above)
Comment (Rkortlandt) [reply src/main/java/frc/robot/RobotContainer.java:144]: F*** You

**Comment by SeanErn** [comment]
Please also resolve the merge conflicts

**Pull request closed without merging**